### ISSUE TITLE
Broken bytecode with secondary ctor and "secondary" properties

### ISSUE THREAD

**Issue by mschirmacher** [Kotlin/kotlinx.serialization#529]
**Describe the bug**

A class with a property declared in the primary constructor, another property declared AND initialized on the classical way and a secondary constructor compiled with kotlin 1.3.31 but does not compile anymore  in 1.3.41

**To Reproduce**
```
import kotlinx.serialization.Serializable

@Serializable
class Test(val something: Boolean) {

    val foo: List<Int> = emptyList()

    constructor() : this(false)
}
```
Fun fact: if `foo` is a primitive type it compiles

**Expected behavior**
It compiles

**Environment**
 - Kotlin version: 1.3.41
 - Library version: 0.11.1
 - Kotlin platforms: JVM, JS
 - Gradle version: 5.4

**Comment by sandwwraith** [comment]
Duplicate of #508? Would be fixed in Kotlin 1.3.50

**Comment by mschirmacher** [comment]
Yes, - works with 1.3.50. Thank you :-)
